core/scripts/Voting.js

```diff
@@ -441,45 +460,67 @@ class VotingSystem {
   }
 
   async runBatchCommit(requests, roundId) {
-    const commitments = [];
+    let commitments = [];
     const skipped = [];
     const failures = [];
+    let batches = 0;
 
-    for (const request of requests) {
-      // Skip commits if a message already exists for this request.
-      // This does not check the existence of an actual commit.
-      if (await this.getMessage(request, roundId)) {
-        skipped.push(request);
-        continue;
-      }
+    // Batch requests up to the max number that we can fit into one block
+    let requestsProcessed = 0;
+    while (requestsProcessed < requests.length) {
+      requestsProcessed += 1;
```

Comment:
ahh great catch. I fixed this, and I added in the actual `batchCommit`/`batchReveal` transactions inside the `if (newCommits/newReveals.length > 0) { ... }` check so that it won't send any empty batch actions